webplugin/js/app/kommunicate-ui.js

```diff
@@ -1803,4 +1803,220 @@ KommunicateUI = {
+                        timeRemainingTimer.text(("0" + minutesElapsed).slice(-2) + ":" + ("0" + secondsElapsed).slice(-2));
+                        $("#wave-front-progressBar").width(percent+'%');
+                        if(REMAINING_TIME <=0){
+                            clearTimeout(playPauseInterval);
+                            pauseBtn.addClass("n-vis");
+                            playBtn.removeClass("n-vis");
+                            hasTimerExpired = true;
+                        }
+                    }else{
+                        START_TIME += 1000;
+                    }
+                }, 1000);
+                timeRemainingTimer.removeClass('n-vis');
+            }
+        };
+        function startRecording() {
+            Kommunicate.typingAreaService.showRecorder();
+            // show
+            kommunicateCommons.modifyClassList(
+                {
+                    id: [
+                        'delete-recording',
+                        'mck-stop-recording',
+                        'audiodiv'
+                    ],
+                },
+                '',
+                'n-vis'
+            );
+
+            // hide
+            kommunicateCommons.modifyClassList(
+                {
+                    id: [
+                        'play-btn',
+                        'pause-btn',
+                        'send-btn',
+                    ],
+                },
+                'n-vis',
+                ''
+            );
+            timeElapsedTimer.removeClass("n-vis");
+            timeRemainingTimer.removeClass("n-vis");
+
+
+            Fr.voice.record(LIVE_OUTPUT, function () {
+                console.log('Recording started');
+                initTimer();
+            }, null,function(err){
+                console.log('error', err);
+                resetRecorder(null, true);
+            });
+        };
+        function initTimer() {
+            var startTime = new Date().getTime();
+            var endTime = startTime + MAX_RECORD_TIME;
+            recorderInterval = setInterval(function () {
+                var now = new Date().getTime();
+                var timeElapsed = now - startTime;
+                var timeRemaining = endTime - now;
+                var percent = timeElapsed / MAX_RECORD_TIME * 100;
+                var secondsElapsed = Math.floor((timeElapsed / 1000) % 60);
+                var secondsRemaining = Math.floor((timeRemaining / 1000) % 60);
+                var minutesElapsed = Math.floor((timeElapsed / (1000 * 60)) % 60);
+                var minutesRemaining = Math.floor((timeRemaining / (1000 * 60)) % 60);
+                timeElapsedTimer.text(("0" + minutesElapsed).slice(-2) + ":" + ("0" + secondsElapsed).slice(-2));
+                timeRemainingTimer.text(("0" + minutesRemaining).slice(-2) + ":" + ("0" + secondsRemaining).slice(-2));
+                $("#wave-front-progressBar").width(percent+'%');
```

Comment:
what's $ here? do we need to use $applozic?

Reply:
same at other places

Reply:
forgot to replace $ with $applozic. Corrected now.